src/classes/BDI_FieldMappingHelpText.cls

```diff
@@ -150,4 +155,183 @@ public with sharing class BDI_FieldMappingHelpText implements BDI_FieldMapping {
+
+            return customMd;
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Wrapper class for Data_Import_Field_Mapping__mdt used to create
+    * the default CMT data as well as for use in the Custom UI.
+    */
+    public class DataImportFieldMapping {
+        public String recordName;
+        public String label {get; set;}
+        public String dataImportObjectName {get; set;}
+        public String dataImportFieldMappingSetRecordName;
+        public String dataImportFieldMappingSetLabel;
+        public String sourceFieldAPIName;
+        public String targetFieldAPIName {get; set;}
+        public String targetObjectMappingRecordName;
+        public String targetObjectMappingLabel;
+        public String targetObjectAPIName;
+        public String isRequired;
+        public Boolean isDeleted;
+        public String importedRecordField;
+        public String importedRecordStatusField;
+
+        public DataImportFieldMapping(String label, String sourceFieldAPIName, String dataImportObjectName,
+            String targetFieldAPIName, String targetObjectAPIName) {
+
+            this.label = label.trim();
+            this.sourceFieldAPIName = sourceFieldAPIName;
+            this.dataImportObjectName = dataImportObjectName;
+            this.targetFieldAPIName = targetFieldAPIName;
+            this.targetObjectAPIName = targetObjectAPIName;
+        }
+
+        public DataImportFieldMapping(Data_Import_Field_Mapping__mdt sourceFieldMapping, String fieldMappingSet) {
+            this.dataImportFieldMappingSetRecordName = fieldMappingSet;
+            this.label = sourceFieldMapping.MasterLabel;
+            this.sourceFieldAPIName = sourceFieldMapping.Source_Field_API_Name__c;
+            this.targetFieldAPIName = sourceFieldMapping.Target_Field_API_Name__c;
+            this.targetObjectMappingRecordName = sourceFieldMapping.Target_Object_Mapping__r.DeveloperName;
+            this.targetObjectMappingLabel = sourceFieldMapping.Target_Object_Mapping__r.MasterLabel;
+            this.dataImportObjectName = sourceFieldMapping.Target_Object_Mapping__r.DeveloperName;
+            this.targetObjectAPIName = sourceFieldMapping.Target_Object_Mapping__r.Object_API_Name__c;
+            this.recordName = sourceFieldMapping.DeveloperName;
+            this.isRequired = 'No';
+            this.isDeleted = false;
+        }
+
+        public void setInstance(Data_Import_Object_Mapping__mdt objectMapping,
+            String fieldMappingSetName) {
+
+            this.targetObjectMappingLabel = objectMapping.Label;
+            this.targetObjectMappingRecordName = objectMapping.DeveloperName;
+            this.dataImportFieldMappingSetRecordName = fieldMappingSetName;
+        }
+
+        public void setInstance(Data_Import_Object_Mapping__mdt objectMapping,
```

Comment:
These setInstance methods can be combined to one with a three param signature that handles null being passed in for the targetFieldAPIName.